### ISSUE TITLE
Don't output trailing commas for flow in arguments array pattern

### ISSUE THREAD

**Issue by vjeux** [prettier/prettier#425]
```js
a = ([s=1,]) => 1
```

throws with the flow parser (but not babylon) https://github.com/facebook/flow/issues/3247

We should workaround by not outputting trailing commas there.

**Comment by vjeux** [comment]
```js
var [s=1,] = 1;
```

still throws.

Also, it requires having a `=` somewhere inside of the `[]` to trigger.

**Comment by iammerrick** [comment]
Not sure if related @vjeux but I'm also seeing prettier create code that won't be parsed by the browser either... Check this out:

```javascript
Object.assign({
  something: 'Cool' 
}, otherFunction())
```

After prettier with trailing commas turned on:

```javascript
Object.assign(
  {
    something: 'Cool',
  },
  otherFunction(),
);
```

That last comma prevents the code from being parsed.

[See Example](https://jlongster.github.io/prettier/#%7B%22content%22%3A%22Object.assign(%7B%5Cn%20%20something%3A%20'Cool'%20%5Cn%7D%2C%20otherFunction())%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3Atrue%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22doc%22%3Afalse%7D%7D)

**Comment by rhengles** [comment]
@iammerrick That will be solved by #641 😃
